src/gui/keys.rs

```diff
@@ -5,6 +5,7 @@ use std::collections::HashSet;
 
 /// Keep track of key states between frames
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
+#[allow(clippy::redundant_pub_crate)]
```

Comment:
This is here because without it Clippy shows a false-positive. It says that `pub(crate)` is unnecessary, but when I listened to Clippy it gave an error. I'm not really sure how `pub(crate)` works since I only ever used just `pub`, so just not touching it was the best option I had in mind.

Reply:
pub(crate) means that this struct is visible to anything inside this crate, but is hidden outside the crate. As noted in my main comment, this lint isn't particularly helpful and shows a lot of false positives. It thinks that because this module is only visible within the crate because it's not used in a public module that this definition is redundant, but that's not strictly true in many cases.